pkg/process/metadata/workloadmeta/extractor.go

```diff
@@ -66,6 +71,14 @@ var (
 		subsystem, "diffs_dropped", "The number of diffs dropped due to channel contention")
 )
 
+// GetSharedWorkloadMetaExtractor returns a shared WorkloadMetaExtractor
+func GetSharedWorkloadMetaExtractor(sysprobeConfig config.Reader) *WorkloadMetaExtractor {
```

Comment:
What makes this shared?

Reply:
Yes, should the naming be different?

Reply:
No, just wasn't obviously to me what this was doing. This can't just be managed by fx right?